Content.Server/Guardian/GuardianSystem.cs

```diff
@@ -0,0 +1,199 @@
+using Content.Shared.Popups;
+using JetBrains.Annotations;
+using Robust.Shared.Audio;
+using Robust.Shared.Containers;
+using Robust.Shared.GameObjects;
+using Robust.Shared.Localization;
+using Robust.Shared.Log;
+using Robust.Shared.Map;
+using Robust.Shared.Player;
+using System;
+using System.Timers;
+
+namespace Content.Server.Guardian
+{
+    public class GuardianSystem : EntitySystem
+    {
+        public override void Initialize()
+        {
+            base.Initialize();
+            SubscribeLocalEvent<GuardianCreatorComponent, UseInHandEvent>(OnGuardianCreated);
+            SubscribeLocalEvent<GuardianCreatorComponent, ExaminedEvent>(OnActivatorExamined);
+            SubscribeLocalEvent<GuardianComponent, MoveEvent>(OnGuardianMove);
+            SubscribeLocalEvent<GuardianHostComponent, MoveEvent>(OnGuardianHostMove);
+            SubscribeLocalEvent<GuardianComponent, DamageChangedEvent>(OnGuardianDamaged);
+            SubscribeLocalEvent<GuardianHostComponent, MobStateChangedEvent>(OnHostStateChange);
+        }
+
+        /// <summary>
+        /// Triggers when the host recives damage which puts in either critical or killed state
+        /// </summary>
+        /// <param name="uid"></param>
+        /// <param name="component"></param>
+        /// <param name="args"></param>
+        private void OnHostStateChange(EntityUid uid, GuardianHostComponent component, MobStateChangedEvent args)
+        {
+            if (EntityManager.GetEntity(component._hostedguardian) != null)
+            {
+                if (args.State.IsCritical())
+                {
+                    var guard = EntityManager.GetEntity(component._hostedguardian);
+                    guard.PopupMessage(Loc.GetString("guardian-host-critical-warn"));
+                    SoundSystem.Play(Filter.Local(), "/Audio/Effects/guardian_warn.ogg", uid);
+                }
+                else if (args.State.IsDead())
+                {
+                    var guard = EntityManager.GetEntity(component._hostedguardian);
+                    SoundSystem.Play(Filter.Pvs(guard), "/Audio/Voice/Human/malescream_guardian.ogg", uid, AudioHelpers.WithVariation(0.25f));
+                    EntityManager.QueueDeleteEntity(guard);
+                    EntityManager.QueueDeleteEntity(uid);
+                }
+            } 
+        }
+
+        private void OnGuardianDamaged(EntityUid uid, GuardianComponent component, DamageChangedEvent args)
+        {
+            var guardiandamage = EntityManager.GetEntity(uid).EnsureComponent<DamageableComponent>();
+            var hostdamage = EntityManager.GetEntity(component.Host).EnsureComponent<DamageableComponent>();
+            if (args.DamageDelta != null)
+            {
+                EntitySystem.Get<DamageableSystem>().SetDamage(hostdamage, (hostdamage.Damage + args.DamageDelta * component.DamagePercent));
+                hostdamage.Owner.PopupMessage(Loc.GetString("guardian-entity-taking-damage"));
+            }
```

Comment:
1. Just use a dependency for the system
2. Use PopupSystem
3. Just call EnsureComponent directly instead of this roundabout GetEntity way

Reply:
Will be fixed